### ISSUE TITLE
About reset messages in parallel nodes

### ISSUE THREAD

**Issue by miccol** [CARVE-ROBMOSYS/BTCompiler#5]
Hi @altacch 

About the [Reset Messages](https://github.com/CARVE-ROBMOSYS/BTCompiler/wiki/Reset-messages) for the parallel node. I think the running children need to be halted whenever the parallel node returns success or failure.

Consider the example `Parallel(T1, T2, T3, N=3)`. As soon as one child T fails, the execution of the other children is no longer needed.

What do you think?

**Comment by altacch** [comment]
Sounds right, it would be the analogue of a "short-circuit" evaluation strategy for the usual Boolean operators. It may be nontrivial to implement it in Coq, though...

**Comment by altacch** [comment]
Reset support for parallel nodes has been added (following the above logic). Of course, some testing is probably needed...

**Comment by altacch** [comment]
I have added some tests for parallel nodes and everything seems to work as expected. 
To everybody: feel free to add more test cases (perhaps with more complicated trees). Notice that the new `test` program will accept multiple input files on the command line.
Don't forget to update the `expected-results` file.
